ext/hiredis_ext/connection.c

```diff
@@ -107,6 +164,26 @@ typedef fd_set _fdset_t;
 #endif
 
 static int __wait_readable(int fd, const struct timeval *timeout, int *isset) {
+#ifdef HAVE_RUBY_FIBER_SCHEDULER_H
+    VALUE scheduler = rb_fiber_scheduler_current();
+    if (scheduler != Qnil) {
+        VALUE result = rb_fiber_scheduler_io_wait(scheduler,
+                                                  io_from_fd(fd),
+                                                  RB_UINT2NUM(RUBY_IO_READABLE),
+                                                  rb_fiber_scheduler_make_timeout((struct timeval *) timeout));
```

Comment:
Should I be using io_from_fd here? Is it going to allocate a Ruby IO object every time it gets to this fn? What's the alternative?

Reply:
The simplest most compatible option is to look up `IO.for_fd` method and call it dynamically. We can introduce a C interface for this, but it won't be supported on older rubies, so you'll need to have that as a fallback.